woocommerce-ac.php

```diff
@@ -4181,18 +4181,18 @@ public function wcal_preview_email_sent() {
 					$body_email_preview   = str_ireplace( '{{coupon.code}}', $coupon_code_to_apply, $body_email_preview );
 				}
 
-				$body_email_preview = str_ireplace( '{{customer.firstname}}', 'John', $body_email_preview );
-				$body_email_preview = str_ireplace( '{{customer.firstname}}', 'John', $body_email_preview );
-				$body_email_preview = str_ireplace( '{{customer.lastname}}', 'Doe', $body_email_preview );
-				$body_email_preview = str_ireplace( '{{customer.fullname}}', 'John Doe', $body_email_preview );
+				$body_email_preview = str_ireplace( '{{customer.firstname}}', apply_filters( 'wcal_abandoned_cart_email_content_customer_firstname', 'John', true, null ), $body_email_preview );
+				$body_email_preview = str_ireplace( '{{customer.firstname}}', apply_filters( 'wcal_abandoned_cart_email_content_customer_firstname', 'John', true, null ), $body_email_preview );
```

Comment:
@nidhitatosaniya - the same line is present twice. Please remove the second occurence